docs/rfcs/0002-storing-course-data.md

```diff
@@ -0,0 +1,134 @@
+## Storing course data
+
+### Abstract
+
+The aim of this RFC is to make decisions about the following:
+- The model definitions* that will be needed to store course data.
+- The plan for loading legacy course data, and keeping that data current
+  after we switch to using our own Open edX instance.
+
+Out of scope for this RFC:
+- Content management – we will just rely on Django admin for creating/updating
+  courses by hand.
+- Any kind of grading/certificate data.
+
+\* - _Much like MicroMasters, we will keep our own models for courses/programs
+  that will exist in parallel with the course data in edX_
+
+### Models
+
+Since this app has very similar needs, `Program`, `Course`, and `CourseRun` 
+models will very closely mirror the [models in MicroMasters](https://github.com/mitodl/micromasters/blob/master/courses/models.py), 
+minus a few fields that are not relevant for this app.
+
+`Program`
+- title
+- description
+- thumbnail (image)
+- live (boolean - indicates whether the program is live in edX, et. al.)
+- merge_key (string - allows us to maintain a reference to a single program
+  that is offered across different courseware platforms. See the section below
+  for more detail.)
+- source (string/enum - indicates where this was loaded/created from)
+
+`Course`
+- program (`Program` foreign key, **nullable** since a course doesn't necessarily belong to a program)
+- position_in_program (integer - indicates course sort order in a program)
+- title
+- description
+- thumbnail (image)
+- live (same as above)
+- merge_key (same as above)
+- source (same as above)
+
+`CourseRun`
+- course (`Course` foreign key)
+- external_id (string - the ID for the course run as it exists in
+  the courseware platform, e.g.: an edX course key like `course-v1:MITxPRO+AMx+1T2019`)
+- external_url (string - the URL for the course on the courseware platform)
+- title
+- start_date
+- end_date
+- enrollment_start
+- enrollment_end
+- live (same as above)
+
+Some things to debate/pay particular attention to:
+
+1. Programs/courses offered across different platforms
+    
+    From the outset we are going to run into a scenario where programs and
+    courses are going to 'switch' platforms from edx.org to our own Open edX
+    instance. Our approach will be to create a separate `Program`/`Course`
+    record when they make this jump, and maintain the same `source_key` value
+    for the records on both platforms so that we will be able to 'join' them 
+    later if needed (example: we may want to show completion progress or payment
+    history for a user that took some program courses on edx.org and other
+    program course on our Open edX instance)
+
+1. **`CourseRun.external_url`**
+    
+    We need to be able to link directly to a course on its courseware
```

Comment:
As I suggested , we can discuss the legacy courses in a different RFC, but what would probably happen is that the day we switch, edx will remove mitxpro.mit.edu and all its content. So there is no link back.  So we just need to import some metadata about mainly course runs to be able to say to the users that "you took a course named X last year, passed and here is the certificate"

Reply:
Then I would suggest we keep the `external_url` field and leave it blank for legacy courses. If we use different courseware for some program in the future, but we want to keep running our own Open edX instance for some courses, we would want to keep this field.